app/jobs/inboxes/fetch_imap_emails_job.rb

```diff
@@ -6,7 +6,8 @@ class Inboxes::FetchImapEmailsJob < MutexApplicationJob
   def perform(channel)
     return unless should_fetch_email?(channel)
 
-    with_lock(::Redis::Alfred::EMAIL_MESSAGE_MUTEX, inbox_id: channel.inbox.id) do
+    key = format(::Redis::Alfred::EMAIL_MESSAGE_MUTEX, inbox_id: channel.inbox.id)
+    with_lock(key) do
```

Comment:
this job can be long running. so sticking to safe side

```suggestion
    with_lock(key, 5.minutes) do
```